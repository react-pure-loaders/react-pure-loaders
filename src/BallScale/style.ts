import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';

const ballScale = keyframes(
  {
    '0%': {
      transform: 'scale(0.0)',
    },
    '100%': {
      opacity: 0,
      transform: 'scale(1.0)',
    },
  },
);

export default (color?: string) => ({
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    animation: `${ballScale} 1s 0s ease-in-out infinite`,
    display: 'inline-block',
    height: '60px',
    width: '60px',
  },
});
