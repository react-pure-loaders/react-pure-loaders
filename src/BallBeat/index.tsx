/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
