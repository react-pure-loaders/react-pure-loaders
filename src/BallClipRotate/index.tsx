/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
