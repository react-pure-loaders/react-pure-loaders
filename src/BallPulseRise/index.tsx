/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
