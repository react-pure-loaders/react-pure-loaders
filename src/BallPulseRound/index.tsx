/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
