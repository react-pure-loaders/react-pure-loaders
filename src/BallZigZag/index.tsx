/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
