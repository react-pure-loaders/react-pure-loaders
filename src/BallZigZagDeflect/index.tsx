/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
