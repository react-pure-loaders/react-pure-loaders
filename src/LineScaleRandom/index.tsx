/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
