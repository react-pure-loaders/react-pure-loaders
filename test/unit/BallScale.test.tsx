import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallScale from '../../src/BallScale';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallScale>', () => {
    afterEach(cleanup);

    test('BallScale should match snapshot', () => {
        const { container } = render(<BallScale loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallScale should have default color', () => {
        const { container } = render(<BallScale loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('BallScale should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallScale color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallScale should have no children', () => {
        const { container } = render(<BallScale loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallScale should have three children', () => {
        const { container } = render(<BallScale loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(2);
    });
});
