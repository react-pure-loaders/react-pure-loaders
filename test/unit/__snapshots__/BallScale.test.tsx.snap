// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallScale> BallScale should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scale(0.0);
    -moz-transform: scale(0.0);
    -ms-transform: scale(0.0);
    transform: scale(0.0);
  }

  100% {
    opacity: 0;
    -webkit-transform: scale(1.0);
    -moz-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }
}

.emotion-0>div {
  border-radius: 100%;
  height: 60px;
  margin: 2px;
  width: 60px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 1s 0s ease-in-out infinite;
  animation: animation-0 1s 0s ease-in-out infinite;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <div />
</div>
`;
